test/DurableOrchestrationClientTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using FluentAssertions;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Moq;
+using Xunit;
+
+namespace WebJobs.Extensions.DurableTask.Tests
+{
+    public class DurableOrchestrationClientTests
```

Comment:
I don't think it's necessary to add unit testing for `DurableOrchestrationClient`. I imagine we already have good coverage elsewhere, and I feel a little uncomfortable about mocking some of the internal implementation details, like `EndToEndTraceHelper` and `DurableTaskExtension`. Can we leave these changes out of the PR for now? The main goal is to enable developers to write mocks for the public surface area, and I think your other changes have already achieved and tested this.

Reply:
Thank you for the detailed feedback here. I will revert these changes based on your feedback. Thank you!